sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -405,8 +405,9 @@ case class MultiplyDTInterval(
   override def left: Expression = interval
   override def right: Expression = num
 
-  override def inputTypes: Seq[AbstractDataType] = Seq(DayTimeIntervalType, NumericType)
-  override def dataType: DataType = DayTimeIntervalType
+  // TODO(SPARK-XXXXX): Multiply day-time intervals with any fields
+  override def inputTypes: Seq[AbstractDataType] = Seq(DayTimeIntervalType(), NumericType)
```

Comment:
We should use `DayTimeIntervalType` to accept all kinds of day time intervals.

Reply:
done